File: app/src/main/java/com/jqk/mydemo/show/showview/BaseActivity.java
package com.jqk.mydemo.show.showview;

import androidx.appcompat.app.AppCompatActivity;
import android.view.KeyEvent;

public class BaseActivity extends AppCompatActivity {
    public ShowView showView;

    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event) {

        if (keyCode == KeyEvent.KEYCODE_BACK
                && event.getRepeatCount() == 0) {
            if (showView != null && showView.isShown()) {
                showView.hideAnim();
                return true;
            }
            return super.onKeyDown(keyCode, event);
        }
        return super.onKeyDown(keyCode, event);
    }
}

File: app/src/main/java/com/jqk/mydemo/base/BaseActivity.kt
package com.jqk.mydemo.base

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity

open class BaseActivity : AppCompatActivity() {

    fun jumpActivity(clazz: Class<*>) {
        var intent = Intent()
        intent.setClass(this, clazz)
        startActivity(intent)
    }
}